### ISSUE TITLE
Create an automated test or CI check to test SQLFluff on popular dbt packages

### ISSUE THREAD

**Issue by barrywhart** [sqlfluff/sqlfluff#2477]
One gap in our automated testing is the lack of a good-sized project with lots of templating.

A partial solution: dbt has a lot of reusable packages (these are SQL/dbt packages, not Python packages). In the same spirit as the recent PR #2473, we should have an automated test or CI check that lints or fixes the **test suites** of some dbt packages. (In effect, their test suites are often very similar to real-life code that SQLFluff users will try and lint.) This could help detect present and future SQLFluff bugs.

* dbt_utils is probably the most popular: https://github.com/dbt-labs/dbt-utils/tree/main/integration_tests
* Others are listed in the dbt package index. Consider including a few of these: https://hub.getdbt.com/

**Comment by infused-kim** [comment]
If you want a real challenge, you could use gitlab's open sourced dbt project:
https://gitlab.com/gitlab-data/analytics/-/tree/master/transform/snowflake-dbt

It has over 2,000 models.

To get it to compile you need to...

#### 1. Add a snowflake profile to `~/.dbt/profiles.yml` and adjust `dbt_project.yml`
```yml
gitlab_test:
  outputs:
    default:
      type: snowflake
      account: test

      # User/password auth
      user: test
      password: test

      role: test
      database: test
      warehouse: test
      schema: test
      threads: 1
      client_session_keep_alive: False
      query_tag: test
```

#### 2. Set some environment variables
```
export SNOWFLAKE_PROD_DATABASE=1
export SNOWFLAKE_PREP_DATABASE=1
export SNOWFLAKE_SNAPSHOT_DATABASE=1
export DATA_TEST_BRANCH=1
export DATA_SIREN_BRANCH=1
export SALT=1
export SALT_PASSWORD=1
export SALT_IP=1
export SALT_EMAIL=1
export SALT_NAME=1
export SNOWFLAKE_LOAD_DATABASE=1
export SNOWFLAKE_TRANSFORM_WAREHOUSE=1
```

#### 3. Delete some models with external dependencies

Edit `packages.yml` and take out:
- git: git@gitlab.com:gitlab-data/data-tests.git
- git: git@gitlab.com:gitlab-data/datasiren.git

Remove files:
```shell 
rm -rf models/legacy/datasiren/
rm -rf models/sources/datasiren/
rm models/workspaces/workspace_data/datasiren/datasiren_audit_results_summary.sql
```

**Comment by barrywhart** [comment]
@infused-kim: Oh, great suggestion! Do you know how long it takes SQLFluff to lint that project?
